bmwqemu.pm

```diff
@@ -305,6 +306,22 @@ sub save_json_file {
     return rename("$fn.new", $fn);
 }
 
+=head2 save_yaml_file
+
+  save_yaml_file([$data, $fn]);
+
+Store validate and save yaml string into the given filename.
+=cut
+sub save_yaml_file {
+    my ($data, $fn) = @_;
+    open(my $fd, ">", "$fn.new");
+    my $yp = YAML::PP->new;
+    my $yaml = $yp->load_string($data);
```

Comment:
Why is the YAML loaded and then dumped again?

Reply:
> Why is the YAML loaded and then dumped again?

validation?

Reply:
That does the validation.